docs/pywbemclicommands.rst

```diff
@@ -166,18 +166,106 @@ Thus, for example an ``instance enumerate`` with and without the ``-o`` option:
+  one set of surrounding double quotes (i.e. they cannot be constructed via
+  string concatenation).
+  Note that charValue when used in INSTANCENAME has exactly one set of
+  surrounding single quotes.
+  Note that DSP0004 prevents the use of real32 or real64 typed properties as
+  keys.
+
+* escaped_INSTANCENAME is a backslash-escaped INSTANCENAME where at
+  least backslash and double quote characters are backslash-escaped
+
+Note that the UNIX-like shells interpret single and double quotes in a certain
+way and remove them before passing the arguments on to the program invoked.
+Because the single and double quotes in INSTANCENAME need to be passed on to
+pywbemcli, they need to be protected from removal by the shell.
+This can be achieved by putting INSTANCENAME into single quotes if it only
+includes double quotes, or into double quotes if it only includes single quotes.
+If there is a mix of single and double quotes in INSTANCENAME, or if shell
+variables need to be expanded, this can be achieved by backslash-escaping any
+double quotes in INSTANCENAME, and putting it into double quotes.
+
+Examples for UNIX-like shells:
+
+.. code-block:: text
+
+   pywbemcli instance get /root/cimv2:MY_Foo.ID=42
+   pywbemcli instance get /MY_Foo.ID=42
+   pywbemcli instance get "/MY_Foo.CharKey='x'"
+   pywbemcli instance get '/MY_Foo.InstanceID="foo1"'
+   pywbemcli instance get "/MY_Foo.InstanceID=\"$value\""
+   pywbemcli instance get '/MY_CS.CreationClassName="MY_CS",Name="MyComp"'
+   pywbemcli instance get '/MY_LogEntry.Timestamp="20190901183853.762122+120"'
+
+Examples for Windows command processor:
+
+.. code-block:: text
+
+   pywbemcli instance get /root/cimv2:MY_Foo.ID=42
+   pywbemcli instance get /MY_Foo.ID=42
+   pywbemcli instance get /MY_Foo.CharKey='x'
+   pywbemcli instance get /MY_Foo.InstanceID="foo1"
+   pywbemcli instance get /MY_Foo.InstanceID="%value%"
+   pywbemcli instance get /MY_CS.CreationClassName="MY_CS",Name="MyComp"
+   pywbemcli instance get /MY_LogEntry.Timestamp="20190901183853.762122+120"
+
+
 .. _`Interactively selecting INSTANCENAME`:
 
 Interactively selecting INSTANCENAME
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Arguments like the INSTANCENAME on some of the instance group commands (
-``get``, ``references``, ``associators``, etc) can be very difficult to correctly enter
-since it can involve multiple keybindings, use of quotation marks, etc.  To
-simplify this pywbemcli includes a option (``-i`` or ``--interactive``) on
-these commands that allows the user to specify only the class name, retrieves
-all the instance names from the server and presents the user with a select list
-from which an instance name can be chosen. The following is an example:
+The INSTANCENAME argument has a certain complexity, particularly for
+associations and for classes with multiple keys.
+
+To simplify this, pywbemcli provides an option (``-i`` or ``--interactive``) on
```

Comment:
What I would really like to do is to drop the interactive and even if it is only a classname, do a get, expecting an error return (just to cover the normally nonexistent singleton instance (i.e. no keys) then if it was just a classname do the select. But that is for later.

Reply:
I don't like that, because this approach cannot distinguish between a user error and the user intention to select interactively.